File: C#/listar-dominios.cs
using System;
using System.Net.Http;
using System.Text;
using System.Threading.Tasks;
using Newtonsoft.Json;

class Program
{
	// Definição do endpoint da API para listar domínios
	private static string url = "https://api.mailgrid.net.br/domain/list/";

	static async Task Main(string[] args)
	{
		// Dados de autenticação (substitua pelo seu token real)
		var data = new
		{
			token_auth = "SEU_TOKEN_AQUI" // Token obtido no painel do cliente
		};

		// Inicializa o HttpClient
		using (var client = new HttpClient())
		{
			// Configura o cabeçalho da requisição para JSON
			client.DefaultRequestHeaders.Add("Content-Type", "application/json");

			// Serializa o objeto de dados para JSON
			var jsonData = JsonConvert.SerializeObject(data);
			var content = new StringContent(jsonData, Encoding.UTF8, "application/json");

			try
			{
				// Envia a requisição POST
				HttpResponseMessage response = await client.PostAsync(url, content);

				// Obtém o código de status HTTP
				int statusCode = (int)response.StatusCode;
				Console.WriteLine($"Código de status: {statusCode}");

				// Verifica se a requisição foi bem-sucedida (código 2xx)
				if (response.IsSuccessStatusCode)
				{
					// Lê a resposta como string
					string responseBody = await response.Content.ReadAsStringAsync();

					// Tenta desserializar a resposta JSON
					var decodedResponse = JsonConvert.DeserializeObject(responseBody);

					// Exibe a resposta formatada em JSON
					Console.WriteLine("Resposta da API:");
					string formattedResponse = 
					JsonConvert.SerializeObject(decodedResponse, Formatting.Indented);
					Console.WriteLine(formattedResponse);
				}
				else
				{
					// Exibe erro caso o código de status seja maior ou igual a 400
					Console.WriteLine($"Erro na requisição. Código de status HTTP: {statusCode}");
				}
			}
			catch (Exception ex)
			{
				// Exibe qualquer erro durante a requisição
				Console.WriteLine($"Erro ao processar a requisição: {ex.Message}");
			}
		}
	}
}
		

File: ruby/listar-dominios.rb
require 'net/http'
require 'json'
require 'uri'

# Definição do endpoint da API
url = URI("https://api.mailgrid.net.br/domain/list/")

# Dados de autenticação (substitua pelo seu token real)
data = {
	"token_auth" => "SEU_TOKEN_AQUI" # Token obtido no painel do cliente
}

# Configuração da requisição HTTP
http = Net::HTTP.new(url.host, url.port)
request = Net::HTTP::Post.new(url.path, {'Content-Type' => 'application/json'})
request.body = data.to_json # Converte os dados para JSON antes de enviar

# Envia a requisição e obtém a resposta
response = http.request(request)

# Verifica o status da resposta
if response.code.to_i >= 400
	# Se o status for maior ou igual a 400, houve erro na requisição
	puts "Erro na requisição. Código de status HTTP: #{response.code}"
else
	# Caso a resposta seja bem-sucedida, decodifica e exibe o JSON
	begin
	decoded_response = JSON.parse(response.body)
	puts "Resposta da API:"
	puts JSON.pretty_generate(decoded_response) # Exibe a resposta formatada
	rescue JSON::ParserError
	# Caso não consiga processar o JSON, exibe um erro
	puts "Erro ao processar a resposta JSON."
	end
end
